DEV_GUIDE.md

```diff
@@ -0,0 +1,163 @@
+
+###Overview
+
+The `scala-exercises` organization holds different repositories for both the core project ("scala-exercises"), and each of the main exercises libraries (i.e.: "exercises-stdlib", "exercises-cats"...). The core project is currently built on one single module, but it'll be potentially splitted up in several ones in a not-so-distant future. The content you can provide to the platform doesn't necessarily be in the previously mentioned organization, but for them to appear inside the main `scala-exercises` in production, they need to be published in a public artifact repository like Sonatype or Maven-central. All the steps, from creating the content library to the publication process, are described in the following sections.
+
+###Content creation
+
+Scala Exercises content is organized in the following way:
+
+```
+Scala Exercises Core
+       |
+       | imports
+       |
+   Libraries
+       |
+       | contains
+       |
+    Sections
+       |
+       | contains
+       |
+    Exercises
+```
+
+####Creating a new content library
+
+Each content library is an independent SBT module working as a library, which should be published as an artifact for it to be included in Scala Exercises. From now on, we'll consider you've already set up a new project for your exercises library, and we'll focus on the specifics on how to layout the content itself.
+
+First of all you need to set up the library by creating a new Scala `object` that contains basic details about it. This `object` must subclass from `org.scalaexercises.definitions.Library`. Here you can find a template for this object:
+
+```scala
+package package_name (1)
+
+import org.scalaexercises.definitions._
+
+/** This line should describe your library shortly, and will appear as its description in the main Scala Exercises website. (2)
+  *
+  * @param name library_name (3)
+  */
+object MyLibrary extends Library {
+  override def owner = "owner" (4)
+  override def repository = "github_repository" (5)
+
+  override def color = Some("#HEX_COLOR") (6)
+
+  (7)
+  override def sections = List(
+    SectionA,
+    SectionB,
+    ...
+    SectionN
+  )
+}
+```
+
+Let's dig into each one of these points:
+
+* (1) A package name for your exercises library. 
+* (2) Inside the first ScalaDoc block, we define the description of the library as it'll appear in the list of the available libraries.
+* (3) Also, this ScalaDoc block defines the title of the library. It'll also be used as the path suffix for its URL. If you need to use spaces, use an underscore symbol (i.e.: "STD_lib")
+* (4) Owner username or organization in GitHub.
```

Comment:
we may want to link to an [Scaladoc syntax reference](https://wiki.scala-lang.org/display/SW/Syntax)